frontend/utilities/helpers.ts

```diff
@@ -617,6 +617,25 @@ export const humanHostDetailUpdated = (detailUpdated?: string): string => {
   }
 };
 
+export const humanHostDiskEncryptionEnabled = (
+  platform?: string,
+  diskEncriptionEnabled?: boolean
+): string => {
+  if (platform === "darwin") {
+    if (diskEncriptionEnabled) {
+      return "The disk is encrypted. The user must enter their<br/> password when they start their computer.";
+    }
+    return "The disk might be encrypted, but FileVault is off. The<br/> disk can be accessed without entering a password.";
+  } else if (platform === "windows") {
+    if (diskEncriptionEnabled) {
+      return "The disk is encrypted. If recently turned on,<br/> encryption could take awhile.";
+    }
+    return "The disk is unencrypted.";
+  }
+  // There is no offstate for Linux as the property will not exist
+  return "Disk encryption is enabled.";
+};
```

Comment:
Can this be restructured to avoid the nested ifs? It's a bit hard to read as-is.

Reply:
Sometimes putting values in a dictionary makes them easier to operate against. How about something like this? 

```typescript
const DISK_ENCRYPTION_MESSAGES = {
  darwin: {
    enabled:
      "The disk is encrypted. The user must enter their<br/> password when they start their computer.",
    disabled:
      "The disk might be encrypted, but FileVault is off. The<br/> disk can be accessed without entering a password.",
  },
  windows: {
    enabled:
      "The disk is encrypted. If recently turned on,<br/> encryption could take awhile.",
    disabled: "The disk is unencrypted.",
  },
};

export const humanHostDiskEncryptionEnabled = (
  platform?: string,
  isDiskEncrypted = false
): string => {
  if (platform !== "windows" && platform !== "darwin") {
    return "Disk encryption is enabled.";
  }
  const encryptionStatus = isDiskEncrypted ? "enabled" : "disabled";
  return DISK_ENCRYPTION_MESSAGES[platform][encryptionStatus];
};
```

Reply:
Love it